src/index.js

```diff
@@ -1,45 +1,12 @@
-import express from 'express';
-import bodyParser from 'body-parser';
-import cors from 'cors';
-import dotenv from 'dotenv';
-import swaggerUI from 'swagger-ui-express';
-import testRoute from './routes/index';
-import docs from '../swagger.json';
+import server from './server';
 
-dotenv.config();
+const PORT = parseInt(process.env.PORT || 4000, 10);
 
-const PORT = process.env.PORT || 4000;
-
-// Create global app object
-const app = express();
-
-// Express config defaults
-app.use(bodyParser.urlencoded({ extended: false }));
-app.use(bodyParser.json());
-app.use(cors());
-
-app.get('/', testRoute);
-
-// route for API documentation
-app.use('/docs', swaggerUI.serve, swaggerUI.setup(docs));
-
-// catch 404 error
-app.use((req, res, next) => {
-  const err = new Error('Not Found');
-  err.status = 404;
-  next(err);
-});
-
-// catch error from database and other errors
-app.use((error, res, next) => {
-  res.status(error.status || 500).json({
-    error: {
-      message: error.message,
-    },
+server
+  .startServer(PORT)
+  .then((app) => {
+    console.log(`Server started on http://localhost:${app.address().port}`);
+  })
+  .catch((err) => {
+    console.error(err.message);
```

Comment:
Great job man! I am impressed with this. Can we use async await for this? We can wrap this in async function then use try catch block. This would allow us to catch more synchronous error. e.g

```

const startUp = async () => {
               try { 
                       await server.startServer(PORT);
                       console.log(`Server started on http://localhost:${app.address().port}`);
               } catch(error) {
                  console.error(err.message);
              }

startUp();
```

Reply:
Okay thanks. I will refactor the code to use async/await